public/timeout.js

```diff
@@ -0,0 +1,161 @@
+var timeoutIntervalId = 0;
+var waitForDOMIntervalId = 0;
+var timeoutGUID = 0;
+var scrollTickerId = 0;
+var sessionLifetime = 1800; //in seconds, default to 30 minutes, modifiable based on access token exp when applicable
+var logoutLocation = "/"; //default logout location, modifiable by config
+var tokenInfo = {};
+
+
+/*
+ * decode Jwt token
+ */
+function parseJwt (token) {
+  var base64Url = token.split('.')[1];
+  var base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
+  var jsonPayload = decodeURIComponent(atob(base64).split('').map(function(c) {
+      return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
+  }).join(''));
+  return JSON.parse(jsonPayload);
+};
+
+/*
+ * get access token information stored in sessionStorage
+ */
+function getSessionTokenInfo() {
+  var keys = Object.keys(sessionStorage);
+  keys.forEach(function(key){
+    var obj;
+    try {
+      obj = JSON.parse(sessionStorage.getItem(key));
+    } catch(e) {
+      obj = null;
+    }
+    if (obj && obj["tokenResponse"] && obj["tokenResponse"]["access_token"]) {
+      tokenInfo = parseJwt(obj["tokenResponse"]["access_token"]);
+    }
+  });
+  console.log("token info? ", tokenInfo);
```

Comment:
This is very helpful for debugging, but probably noisy otherwise. Would it be possible to only enable when SYSTEM_TYPE is not `production`?

Reply:
thank you, Ivan.  I will do that in a separate PR.